data-raw/pull_environmental_data.R

```diff
@@ -0,0 +1,683 @@
+                gage_agency = "USGS",
+                gage_number = "11421000",
+                parameter = "flow",
+                statistic = "mean"
+         )))
+
+# Do a few additional flow data pull tests to confirm that new data pull has 
+# more data
+try(if(nrow(yuba_daily_flows) < nrow(yuba_existing_flow)) 
+  yuba_daily_flows <- yuba_existing_flow)
+
+yuba_river_existing_temp <- SRJPEdata::environmental_data |> 
+  filter(gage_agency == "CDEC" &
+           gage_number == "YR7" &
+           parameter == "temperature") |> 
+  select(-site)
+
+### Temp Data Pull - 
+#Interpolation pull for Yuba
+yuba_river_interpolated <- read_csv(here::here("data-raw", "temperature-data", "yuba_temp_interpolation.csv")) |> 
+  select(-subsite, -site_group, -site) |> 
+  mutate(parameter = "temperature") |> 
+  glimpse()
+
+
+
+yuba_river_existing_temp <- SRJPEdata::environmental_data |> 
+  filter(gage_agency == "CDEC" &
+           gage_number == "YR7" &
+           parameter == "temperature") |> 
+  select(-site)
+
+
+#pull temp data from CDEC YR7
+try(yuba_river_temp_query <- cdec_query(station = "YR7", dur_code = "E", sensor_num = "146", start_date = "2019-01-01"))
+
+# Confirm data pull did not error out, if does not exist - use existing flow, 
+# if exists - reformat new data pull           
+
+try(if(!exists("yuba_river_temp_query"))
+  yuba_river_daily_temp <- yuba_river_existing_temp
+  else({yuba_river_daily_temp <- yuba_river_temp_query |> 
+         mutate(date = as_date(datetime)) |> 
+         mutate(year = year(datetime)) |> 
+         group_by(date) |> 
+         summarise(mean= mean(parameter_value, na.rm = TRUE),
+                   max = max(parameter_value, na.rm = TRUE),
+                   min = min(parameter_value, na.rm = TRUE)) |> 
+         pivot_longer(mean:min, names_to = "statistic", values_to = "value") |>
+         mutate(stream = "Yuba River",
+                gage_agency = "CDEC",
+                gage_number = "YR7",
+                parameter = "temperature") |> 
+  bind_rows(yuba_river_interpolated)
+  } ))
+
+
+### Temp Data Pull Tests 
+
+#Combine all flow data from different streams
+all_flow <- bind_rows(battle_creek_daily_flows,
+                      butte_creek_daily_flows, 
+                      clear_creek_daily_flows,
+                      deer_creek_daily_flows,
+                      feather_hfc_river_daily_flows,
+                      feather_lfc_river_daily_flows,
+                      lower_feather_river_daily_flows,
+                      feather_lfc_river_daily_flows,
+                      mill_creek_daily_flows,
+                      sac_river_daily_flows,
+                      yuba_river_daily_flows) |> 
+  glimpse()
+
+ggplot(all_flow |> 
+         filter(statistic == "mean"),
+         aes(x= date, y = value, color=stream)) +
+  geom_line() +
+  facet_wrap(~stream)
+
+#Combine all temperature data from different streams
+all_temps <- bind_rows(battle_creek_daily_temp,
```

Comment:
Stream name is inconsistent across interpolated and gage sources. Some are lower case, others are upper, and some are a mix. We only want one naming convention for each stream. We can make them all lower case for consistency.

Reply:
changed stream names to be consistent throughout the script to all lower case